test/unit/host_test.rb

```diff
@@ -669,21 +669,38 @@ def teardown
       end
     end
 
-    test "should save if owner_type is User or Usergroup" do
-      host = Host.new :name => "myfullhost", :mac => "aabbecddeeff", :ip => "3.3.4.03",
-        :ptable => FactoryGirl.create(:ptable, :operatingsystem_ids => [operatingsystems(:redhat).id]), :medium => media(:one),
-        :domain => domains(:mydomain), :operatingsystem => operatingsystems(:redhat), :subnet => subnets(:two), :puppet_proxy => smart_proxies(:puppetmaster),
-        :architecture => architectures(:x86_64), :environment => environments(:production), :managed => true,
-        :owner_type => "User", :root_pass => "xybxa6JUkz63w"
-      assert host.valid?
-    end
+    context 'owner_type validations' do
+      test "should save if owner_type is User or Usergroup" do
+        host = FactoryGirl.build(:host, :owner_type => "User", :owner => User.current)
+        assert_valid host
+      end
 
-    test "should not save if owner_type is not User or Usergroup" do
-      host = Host.new :name => "myfullhost", :mac => "aabbecddeeff", :ip => "3.3.4.03", :medium => media(:one),
-        :domain => domains(:mydomain), :operatingsystem => operatingsystems(:redhat), :subnet => subnets(:two), :puppet_proxy => smart_proxies(:puppetmaster),
-        :architecture => architectures(:x86_64), :environment => environments(:production), :managed => true,
-        :owner_type => "UserGr(up" # should be Usergroup
-      assert !host.valid?
+      test "should not save if owner_type is not User or Usergroup" do
+        host = FactoryGirl.build(:host, :owner_type => "UserGr(up") # should be Usergroup
+        refute_valid host
+      end
+
+      test 'should succeed validation if owner not set' do
+        # don't want to use FactoryGirl here, since I am not sure that the owner
```

Comment:
It's not set, check `/test/factories/host_related` or inspect it with pry, please change this